lightning/src/ln/channelmanager.rs

```diff
@@ -2406,14 +2413,17 @@ impl<Signer: Sign, M: Deref, T: Deref, K: Deref, F: Deref, L: Deref> ChannelMana
 					};
 					let (chan_update_opt, forwardee_cltv_expiry_delta) = if let Some(forwarding_id) = forwarding_id_opt {
 						let chan = channel_state.as_mut().unwrap().by_id.get_mut(&forwarding_id).unwrap();
-						// Leave channel updates as None for private channels.
-						let chan_update_opt = if chan.should_announce() {
-							Some(self.get_channel_update_for_unicast(chan).unwrap()) } else { None };
+						let chan_update_opt = self.get_channel_update_for_broadcast(chan).ok();
```

Comment:
I think you should write better the rational for why we're using `get_channel_update_broadcast` instead of unicast. I didn't get the why at first read.

Something like "SCID alias aims to mask the real channel SCID from the payment sender. If we return the real channel SCID in a channel update error message, it would allow a payment sender to deanonymize the alias by triggering HTLC failures on the routing fees or HTLC minimum amount".

Reply:
Good catch, but this doesn't need a comment, this is just wrong!

Reply:
Also updated the test to hit this case.

Reply:
I thought if `short_channel_id` is the real scid for a private channel, we still want to leave the update as `None`?

Reply:
We do - the `chan.get_channel_type().supports_scid_alias() && *short_channel_id != chan.outbound_scid_alias()` check below the `chan_update_opt` get breaks with `None`, dropping the `chan_update_opt`. For clarity I moved the `get_channel_update_for_onion` call down a few lines.

Reply:
Missing something here -- if we're attempting to forward over a private channel without an alias, and we break on e.g. `if !chan.is_live()` 10 lines down, won't we include the private channel update with the current code?

Reply:
Hmm, so the `SCIDAlias` feature is supposed to be the "never, ever, ever tell anyone the real SCID" flag (maybe we should rename it scid_privacy in LDK?). I guess in principle we could treat any private channel as an SCIDPrivacy/SCIDAlias channel for the purposes of `channel_update` generation, but I think eventually after people upgrade a private channel without the flag is rare/indicates some strange use-case where they're gonna use the real SCID for something.